### PR TITLE
Adds Fletcher Mirco Establishment Inside goodsAndServices.js

### LINKED ISSUE #80: Fletcher Creation
Create a new Micro Establishment for a Fletcher inside of goodsAndServices.js using the existing Bakery template.

These buildings should have at least 3 items for sale, at least 5 choices in each list and be fleshed out enough that others could easily contribute to the random lists at a later date.

### PR DESCRIPTION
**Pull request by underdorff18**
Adds fletcher establishment to goodsAndServices.js. 
Fixes #80

### CODE REVIEW

File: EssentialEstablishmentGenerator/Buildings/goodsAndServices.js [hunk 1]
```diff
@@ -1398,6 +1398,289 @@ setup.goodsAndServices = {
...
+        cost: 1000,
+        description: 'A lightweight ranged weapon with little damage. It is often used to deliver poisons.'
+      },
+      'sling': {
+        cost: 10,
+        description: 'A simple and light ranged weapon, used to hurl lead bullets at enemies.'
+      },
+      'light crossbow': {
+        cost: 2500,
+        description: 'A lightweight crossbow, comarable in range to a shortbow. Slow to reload, but packs a punch.'
```
Comment (WishingOnAWendy) [inline EssentialEstablishmentGenerator/Buildings/goodsAndServices.js]: Comparable*

File: EssentialEstablishmentGenerator/Buildings/goodsAndServices.js [hunk 2]
```diff
@@ -1398,6 +1398,289 @@ setup.goodsAndServices = {
...
+          }
+        }
+      },
+      '50 blowgun needles': {
+        cost: 100,
+        description: 'A bundle of 50 blowgun needles.'
+      },
+      '20 sling bullets': {
+        cost: 4,
+        description: 'Small lead ellets used with a sling to bludgeon enemies.'
```
Comment (WishingOnAWendy) [inline EssentialEstablishmentGenerator/Buildings/goodsAndServices.js]: bullets*

File: EssentialEstablishmentGenerator/Buildings/goodsAndServices.js [hunk 3]
```diff
@@ -1398,6 +1398,289 @@ setup.goodsAndServices = {
...
+    },
+    type: 'fletcher',
+    notableFeature: [
+      // you notice _______
+      'there are feathers floating around the room.',
+      'there are pieces of feather strewn about on the floor.',
+      'the smell of fresh cut wood permeates the air.',
+      'an arrow on display with an array of colorful feathers.',
+      'a large bear pelt hung proudly on the wall behind the counter.',
+      'a collection of deer antlers mounter across the wall furthest from you.',
```
Comment (WishingOnAWendy) [inline EssentialEstablishmentGenerator/Buildings/goodsAndServices.js]: mounted*

File: EssentialEstablishmentGenerator/Buildings/goodsAndServices.js [hunk 4]
```diff
@@ -1398,6 +1398,289 @@ setup.goodsAndServices = {
...
+      'the faint smell of beer filling the air.',
+      'the building is ket very warm by a crackling fireplace.',
```
Comment (WishingOnAWendy) [inline EssentialEstablishmentGenerator/Buildings/goodsAndServices.js]: kept*

File: EssentialEstablishmentGenerator/Buildings/goodsAndServices.js [hunk 5]
```diff
@@ -1398,6 +1398,289 @@ setup.goodsAndServices = {
...
+      // the fletcher is known for _______
+      'excellent and quick service.',
+      'giving the best hunting tips.',
+      'knowing the best places to find game.',
+      'gossiping about local drama.',
+      'giving a free lesson with every bow purchase.',
+      'providing the most affordable products around.',
+      'giving the fastest service in town.',
+      'always serving with a smile.',
+      'having the sharest arrows in town.',
```
Comment (WishingOnAWendy) [inline EssentialEstablishmentGenerator/Buildings/goodsAndServices.js]: sharpest*

File: EssentialEstablishmentGenerator/Buildings/goodsAndServices.js [hunk 6]
```diff
@@ -1398,6 +1398,289 @@ setup.goodsAndServices = {
...
+      'crafting arrows that always hit their mark.',
+      'their extremely strong and reliable arrows.',
+      'creating an arrow that can pierce clean through a hog.',
+      'their quivers that keep arrows from falling out when on the move.',
+      'their bowstrings that never snap.',
+      'crafting tough and reliable bows that rarely break.',
+      'offering a free quiver of arrows with every bow purchase.',
+      'making the heaviest bow in the continent.',
```
Comment (WishingOnAWendy) [inline EssentialEstablishmentGenerator/Buildings/goodsAndServices.js]: heaviest bows on the continent*

File: EssentialEstablishmentGenerator/Buildings/goodsAndServices.js [hunk 7]
```diff
@@ -1398,6 +1398,289 @@ setup.goodsAndServices = {
...
+      },
+      '20 bolts': {
+        cost: 100,
+        description: 'A bundle of 20 standard crossbow bolts.'
+      },
+      '10 silvered bolts': {
+        cost: 10050,
+        description: 'Ten bolts plated with silver to bypass weapon immunities. Good for vampires and lycanthropes.',
+        exclusions (town, building) {
+          if (town.population > 2000 && building.roll.wealth > 60) {
```
Comment (Glinkis) [inline EssentialEstablishmentGenerator/Buildings/goodsAndServices.js]: Don't explicitly return `true` and `false`.
```js
return town.population...
```
Is equivalent, and less verbose.

**Comment by ryceg** [comment]
Hi @underdorff18, after some discussion, we are thinking that we are going to change the license of the generator from GPL3 to MIT, in order to make it more accessible, and also stop the "infectious" license from spreading- this way, we will be able to integrate the generator into third party applications, and let people use the generator how they wish. 

However, in order to do this, we need to obtain permission from everyone who has contributed to the project, since you contributed under the GPL3 license, that doesn't mean that you would necessarily be okay with changing to the MIT license. 

You are under no obligation to provide license to use your code under the MIT license- if you do not agree with the license change then we can remove your code and make a fork of the generator without it, just let me know.

Thanks!

Rhys

**Pull request closed without merging**